core/src/main/scala/org/apache/spark/serializer/KryoSerializer.scala

```diff
@@ -307,7 +307,7 @@ private[spark] class KryoSerializerInstance(ks: KryoSerializer) extends Serializ
   override def deserialize[T: ClassTag](bytes: ByteBuffer): T = {
     val kryo = borrowKryo()
     try {
-      input.setBuffer(bytes.array)
+      input.setBuffer(bytes.array(), bytes.arrayOffset() + bytes.position(), bytes.remaining())
```

Comment:
Not necessary for this change, but at some point it might be worth it to change this to use Kryo's `ByteBufferInput`.